Lycanroc rework

Lilypaddo opened the pull request:
Split Lycanroc into 3 Pokemon, Lycanroc, Rougaroc and Wolferoc.

HARRYING STRIKES, Lycanroc signature, increases damage by 50% when alternating between biting and non-biting attacks.
UNNATURAL RESISTANCE, Rougaroc signature, reduces damage taken from non-steel moves by 35% in moonglow.
CANID CRUSHER, Wolferoc signature, applies fractures on the first attack.
STORMSHARDS, signature to all 3, is a 50bp rock move that applies a damaging effect to the targeted slot for 3 turns.

Lilypaddo committed:
Fix vertical tv event

Changed event in novo town geodude house to have proper collision

Lilypaddo committed:
Merge branch 'development' of https://github.com/Pokemon-Tectonic-Team/Pokemon-Tectonic into development

Lilypaddo committed:
Merge branch 'development' of https://github.com/Pokemon-Tectonic-Team/Pokemon-Tectonic into development

Lilypaddo committed:
Merge branch 'development' of https://github.com/Pokemon-Tectonic-Team/Pokemon-Tectonic into development

Lilypaddo committed:
Merge branch 'development' of https://github.com/Pokemon-Tectonic-Team/Pokemon-Tectonic into development

Lilypaddo committed:
Merge branch 'development' of https://github.com/Pokemon-Tectonic-Team/Pokemon-Tectonic into development

Lilypaddo committed:
Merge branch 'development' of https://github.com/Pokemon-Tectonic-Team/Pokemon-Tectonic into development

Lilypaddo committed:
Merge branch 'development' of https://github.com/Pokemon-Tectonic-Team/Pokemon-Tectonic into development

Lilypaddo committed:
Merge branch 'development' of https://github.com/Pokemon-Tectonic-Team/Pokemon-Tectonic into development

Lilypaddo committed:
Merge branch 'development' of https://github.com/Pokemon-Tectonic-Team/Pokemon-Tectonic into development

Lilypaddo committed:
Merge branch 'development' of https://github.com/Pokemon-Tectonic-Team/Pokemon-Tectonic into development

Lilypaddo committed:
Initial Work

Seperate LYCANROC formes into ROUGAROC and WOLFEROC, with seperate statlines and typings.
Change Sprites to accomodate
Add movepools, change existing trainers to accomodate
Implement CANID CRUSHER and UNNATURAL RESISTANCE
Add HARRYING STRIKES and STORMSHARDS, but without function right now

Lilypaddo committed:
Harrying Strikes, minor tweaks

Implemented function for HARRYING STRIKES
Tweaked sprite metrics
Added dex entires
Added Counter to Rougaroc

Lilypaddo committed:
Stormshards

Introduce STORMSHARDS, signature to the Rockruff evos.
Also slight wording change on HARRYING STRIKES

AlphaKretin committed:
Merge branch 'development' into Lycanroc

Update to latest dev, I'm gonna test on this branch in my new playthrough until it's merged

AlphaKretin committed:
Cut UNDERMINE from Rockruff

Now that the TTSI rework has been merged in

Lilypaddo committed:
Minor tweaks to Rougaroc and Rockruff

Renamed Unnatural Resistance to Umbral Tenacity
Gave Rockruff +10 defense to match its attack stat
Changed Vital Spirit on Rockruff to Stubborn
Added Run Amuck, Long Shot and Fighting Chance to Rockruff
Added Plan Ahead to Wolferoc

zinniagamedev commented on PBS/trainers.txt:
```diff
@@ -2220,10 +2219,9 @@ Pokemon = FEAROW,40
-    AbilityIndex = 0 # Tough Claws
+Pokemon = WOLFEROC,40
+    Moves = ADAMANTINEPRESS,DISTURB,STRAFE,ROCKTOMB
+    AbilityIndex = 0 # Stubborn
```
should this perhaps use the other ability?

zinniagamedev commented on PBS/regionaldexes.txt:
```diff
@@ -418,7 +418,7 @@ GRUBBIN,CHARJABUG,VIKAVOLT
 ORICORIO
 CUTIEFLY,RIBOMBEE
-ROCKRUFF,LYCANROC
+ROCKRUFF,LYCANROC,ROUGAROC,WOLFEROC
```
Thinking it might make sense for the order to be LYCANROC,WOLFEROC,ROUGAROC instead

AlphaKretin replied:
I could see it either way - WOLFEROC in the middle because it's dawn, or WOLFEROCK last because it's the weird extra one tacked on later. But I'm happy to make the change, especially since the latter is mostly due to Canon Jank that we've cut out.

zinniagamedev commented on PBS/trainers.txt:
```diff
@@ -8199,8 +8197,8 @@ Pokemon = SAMUROTT,70
 Pokemon = LYCANROC,70
-    Form = 2 # Dusk Form
-    AbilityIndex = 0 # Tough Claws
+    Form = 2 #  Form
```
Remove the Form definitions from these trainer entries.

zinniagamedev commented on PBS/trainers_monument.txt:
```diff
@@ -432,10 +432,9 @@ Pokemon = ASANDSLASH,70
-    AbilityIndex = 0 # Tough Claws
+Pokemon = WOLFEROC,70
+    Moves = ADAMANTINEPRESS,POWDERBOMB,STRAFE,FLAMETHROWER
+    AbilityIndex = 0 # Stubborn
```
This also looks like it should use the other ability.

zinniagamedev commented on PBS/trainers.txt:
```diff
@@ -8259,8 +8257,8 @@ Policies = CURSE_STORM_TOTEM
     Item = CHARTIBERRY
-Pokemon = LYCANROC,70
-    Moves = BEDROCKBREAKER,HUMBLE,STOMPINGTANTRUM,CLOSECOMBAT
+Pokemon = WOLFEROC,70
```
What ability is this one using? Also, its slightly sad that no trainer seems to use ROUGAROC currently.

AlphaKretin replied:
This one's a cursed fight inheriting from the entry above - which is another one using Stubborn that should probably be switched over, so I'll do that. As for ROUGAROC, currently working on adding that to a trainer in the thread.

zinniagamedev commented on PBS/abilities_new.txt:
```diff
@@ -904,6 +908,10 @@ Description = If an attack lowers the Pokémon's HP below half, the attacker bec
 #-------------------------------
+[HARRYINGSTRIKES]
+Name = Harrying Strikes
+Description = Biting attacks deal 50% more damage if the last move was a non-biting attack, and vice versa.
```
Sad that it learns BARETEETH by level up but the ability doesn't actually work with it.

AlphaKretin replied:
Should we just change Harrying Strikes to be (non)-"biting moves" instead of attacks? Or is the issue that we'd want BARETEETH to be boostable somehow too?

zinniagamedev commented on Plugins/Tectonic Battle/Effects/EffectDefinitions/PositionEffects.rb:
```diff
@@ -242,4 +242,26 @@
     end,
+})
+
+GameData::BattleEffect.register_effect(:Position, {
```
Needs an apply_proc announcing when it is applied.

zinniagamedev commented on Plugins/Tectonic Battle/Abilities/Ability Event Handlers/Damage/DamageCalcTargetAbilities.rb:
```diff
@@ -410,4 +410,13 @@
   }
+)
+
+BattleHandlers::DamageCalcTargetAbility.add(:UNNATURALRESISTANCE,
```
Wasn't changed to the new ability name UMBRALTENACITY.

AlphaKretin committed:
Update WOLFEROC ability on trainer teams

AlphaKretin committed:
Remove outdated Form parameters for LYCANROC trainers

ShadowKirbae commented:
Evolutions = LYCANROC,AttackGreater,27,ROUGAROC,DefenseGreater,27,WOLFEROC,AtkDefEqual,27
BaseStats = 75,100,85,125,60,75 (Lycanroc)
BaseStats = 75,110,75,95,110,75 (Wolferoc)

I think it might make sense to make it so Rockruff evolves into Lycanroc when Atk and Def are equal, rather than Wolferoc, since the difference between the stats is quite a bit higher.

AlphaKretin committed:
Update UMBRALTENACITY internal name

AlphaKretin commented:
> Evolutions = LYCANROC,AttackGreater,27,ROUGAROC,DefenseGreater,27,WOLFEROC,AtkDefEqual,27 BaseStats = 75,100,85,125,60,75 (Lycanroc) BaseStats = 75,110,75,95,110,75 (Wolferoc)
> 
> I think it might make sense to make it so Rockruff evolves into Lycanroc when Atk and Def are equal, rather than Wolferoc, since the difference between the stats is quite a bit higher.

Flavourfully, Wolferoc is the Dawn form which requires a balance between day and night, so I definitely think it should be the AtkDefEqual evolution. It might be worth adjusting the stats to reflect this?

AlphaKretin committed:
Add apply_proc to Stormshards effect

AlphaKretin committed:
Update dex order for ROCKRUFF line

AlphaKretin committed:
Update WOLFEROC ability for trainer TAPUKOKO

AlphaKretin committed:
Add ROUGAROC to Swimmer Stannis

Replacing Lunatone, since Mola + Lunatone is already used as a Moonglow core on another trainer

AlphaKretin committed:
Rework and rename HARRYINGSTRIKES

Allow non-attack moves to trigger it, but only works one way

AlphaKretin committed:
Fix TOOTHANDCLAW name
